testing/is_contiguous_iterator.cu

```diff
@@ -134,3 +134,89 @@ void test_is_contiguous_iterator_vectors()
 }
 DECLARE_VECTOR_UNITTEST(test_is_contiguous_iterator_vectors);
 
+template <typename IteratorT, typename PointerT, bool ExpectPointer>
+struct check_unwrapped_iterator
+{
+  using unwrapped_t = typename std::remove_reference<
+    decltype(thrust::detail::try_unwrap_contiguous_iterator(
+      std::declval<IteratorT>()))>::type;
+
+  using result =
+    typename std::conditional<ExpectPointer,
+                              std::is_same<unwrapped_t, PointerT>,
+                              std::is_same<unwrapped_t, IteratorT>>::type;
+
+  static constexpr bool value = result::value;
+};
+
+template <typename T>
+void test_try_unwrap_contiguous_iterator()
+{
+  THRUST_STATIC_ASSERT((check_unwrapped_iterator<T *,
+                                                 T *,
+                                                 true>::value));
+  THRUST_STATIC_ASSERT((check_unwrapped_iterator<T *,
+                                                 T *,
+                                                 false>::value));
```

Comment:
Since the unwrapped raw pointer is the same type as the pass-through, both the `true` and `false` cases should pass here. Testing both just ensures that everything is working as expected.

Reply:
Since the unwrapped raw pointer is the same type as the pass-through, both the `true` and `false` cases should pass here. Testing both just ensures that everything is working as expected.